dnsapi/dns_ispconfig.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env sh
+
+#ISPConfig 3.1 API - Add remote user and give him access to at least the "DNS txt functions"
+
+# User must provide login data and URL to the ISPConfig installation incl. port. The remote user in ISPConfig must have access to:
+# - DNS zone Functions
+# - DNS txt Functions
+
+# Values to export:
+# export ISPC_User="remoteUser"
+# export ISPC_Password="remotePasword"
+# export ISPC_Api="https://ispc.domain.tld:8080/remote/json.php"
+
+########  Public functions #####################
+
+#Usage: dns_myapi_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_ispconfig_add() {
+  fulldomain="${1}"
+  txtvalue="${2}"
+  _ISPC_credentials && _ISPC_login && _ISPC_getZoneInfo && _ISPC_addTxt || return 1
+}
+
+#Usage: dns_myapi_rm   _acme-challenge.www.domain.com
+dns_ispconfig_rm() {
+  fulldomain="${1}"
+  _ISPC_login && _ISPC_rmTxt || return 1
+}
+
+####################  Private functions bellow ##################################
+
+_ISPC_credentials() {
+  if [ -z "$ISPC_User" ] || [ -z "$ISPC_Password" ] || [ -z "$ISPC_Api" ]; then
+    ISPC_User=""
+    ISPC_Password=""
+    ISPC_Api=""
+    _err "You haven't specified the ISPConfig Login data and the URL. Please try again."
+    return 1
+  else
+    _saveaccountconf ISPC_User "${ISPC_User}"
+    _saveaccountconf ISPC_Password "${ISPC_Password}"
+    _saveaccountconf ISPC_Api "${ISPC_Api}"
+  fi
+}
+
+_ISPC_login() {
+  _info "Getting Session ID"
+  curData="{\"username\":\"${ISPC_User}\",\"password\":\"${ISPC_Password}\",\"client_login\":false}"
+  curResult=$(_post "${curData}" "${ISPC_Api}?login")
```

Comment:
I suggest your add another option for your api:

```
ISPC_User 
ISPC_Password 
ISPC_Api 

ISPC_Api_Insecure
```

if the user set `ISPC_Api_Insecure` then you set `HTTPS_INSECURE=1`,  otherwise do not set.